### PR TITLE
Issue/5113 create quick order

### LINKED ISSUE #5113: [Quick Order] Create order from Quick Order view
When the user taps Done in the quick order view, we need to create a new order. Based on [the iOS code](https://github.com/woocommerce/woocommerce-ios/blob/1e4e1e6ed42f5e2da7db679573dc276b5c8b673f/Yosemite/Yosemite/Stores/OrderStore.swift#L250-L264), the only field we need to pass is `fee_lines`.

[Related FluxC issue](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2183)

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #5113 by creating the quick order from the quick order view, and navigating to the newly created order detail.
 
To test:

* Make sure "Quick order" is enabled in Settings > Beta features
* Go to the order list
* Tap the + icon
* Enter an amount
* Tap Done
* Ensure you're taken to the order detail
* Ensure order is created with the correct amount

view

### CODE REVIEW

**Comment by ThomazFB** [comment]
I assigned myself to this PR just to be aware of any updates more easily, just let me know when it's ready to review and I'll jump to review it right away!

**Comment by nbradbury** [comment]
@ThomazFB This is ready for review

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/quickorder/QuickOrderViewModel.kt [hunk 1]
```diff
@@ -37,10 +49,42 @@ class QuickOrderViewModel @Inject constructor(
...
+            return
+        }
+
+        viewState = viewState.copy(isProgressShowing = true, isDoneButtonEnabled = false)
+
+        launch {
+            val result = orderStore.postQuickOrder(
+                selectedSite.get(),
+                viewState.currentPrice.toString()
+            )
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/quickorder/QuickOrderViewModel.kt]: we're not shifting the launch context here, so this request will end up getting triggered inside a coroutine of `Dispatchers.Main` already, I suggest we launch it using the `dispatchers.io` through injection of the `CoroutineDispatchers` like this:
```suggestion
               launch(dispatchers.io) {
            val result = orderStore.postQuickOrder(
                selectedSite.get(),
                viewState.currentPrice.toString()
            )
```

Or we could just remove the `withContext(Dispatchers.Main)` for the following code, but we would still be triggering the request inside a Main dispatcher coroutine, which could lead us to ANRs. WDYT?

**Review by ThomazFB — COMMENTED** [review]
The feature is working as expected, all orders submitted to the site went well. Awesome work here! I left just one comment about how we're launching the request that might lead to some ANRs.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/quickorder/QuickOrderViewModel.kt (hunk 1 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/quickorder/QuickOrderViewModel.kt]: @ThomazFB Thanks for the tip, I'm still wrapping my head around coroutine contexts so any advice is welcome. I made the suggested change in da9dc5a.

**Review by ThomazFB — APPROVED** [review]
Thanks for applying the suggestions, @nbradbury! LGTM :shipit:

**Pull request merged**